### PR TITLE
🚀 Add icon font linter rule

### PR DESCRIPTION
**Pull request by lluerich**
This checks if commonly used classnames are being used on a page and checks further for known font-families of icon fonts in the ```@font-face``` declarations in the ```<style amp-custom>``` element.

### CODE REVIEW

File: packages/linter/package.json [hunk 1]
```diff
@@ -21,6 +21,8 @@
     "cross-fetch": "3.0.6",
     "debug": "4.3.1",
     "execa": "5.0.0",
+    "postcss": "^8.2.6",
```
Comment (sebastianbenz) [inline packages/linter/package.json:24]: please pin versions

File: packages/linter/src/rules/NoIconFontIsUsed.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            }
+          }
+        });
+      };
+    });
+
+    // grab the amp-custom css
+    const stylesText = $("style[amp-custom]").html();
+
+    if (stylesText) {
```
Comment (sebastianbenz) [inline packages/linter/src/rules/NoIconFontIsUsed.ts:78]: return early instead

File: packages/linter/src/rules/NoIconFontIsUsed.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          console.warn(`Failed to process CSS`, err.message);
+          return { css: stylesText };
+        });
+    }
+
+    if (iconFontMatches.length > 0) {
+      return this.fail("It seems like icon fonts are being used on this page.");
+    }
+
+    return this.warn("Suspicious icon font class names detected.");
```
Comment (sebastianbenz) [inline packages/linter/src/rules/NoIconFontIsUsed.ts:94]: How likely is this case?

File: packages/linter/src/rules/NoIconFontIsUsed.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import { Context } from "../index";
+import { Rule } from "../rule";
+
+const postcss = require("postcss");
+const safeParser = require("postcss-safe-parser");
+
+const ICON_FONT_IDENTIFIERS = [
+  {
+    className: "fa-",
```
Comment (sebastianbenz) [inline packages/linter/src/rules/NoIconFontIsUsed.ts:9]: Do we even need to match by classname? Or is it common that icon font names are changed?

**Review by sebastianbenz — APPROVED** [review]
Looks good!

File: packages/linter/src/rules/NoIconFontIsUsed.ts (hunk 3 above)
Comment (lluerich) [reply packages/linter/src/rules/NoIconFontIsUsed.ts:94]: Well so far I know of one case, concerning the Material Icons Font. The stylesheet is loaded via Google Web Fonts, so there is no custom css to scan for font-families, only classnames. So in this case we might give out a warning.

File: packages/linter/src/rules/NoIconFontIsUsed.ts (hunk 4 above)
Comment (lluerich) [reply packages/linter/src/rules/NoIconFontIsUsed.ts:9]: As mentioned above, there are some cases that use an external stylesheet via Google Web Fonts, and a way to identify those, would be to match by classname.

File: packages/linter/package.json (hunk 1 above)
Comment (lluerich) [reply packages/linter/package.json]: 👍🏻

File: packages/linter/src/rules/NoIconFontIsUsed.ts (hunk 3 above)
Comment (sebastianbenz) [reply packages/linter/src/rules/NoIconFontIsUsed.ts:94]: In that case we could also check for `<link href="https://fonts.googleapis.com/icon?family=Material+Icons"
      rel="stylesheet">`

File: packages/linter/src/rules/NoIconFontIsUsed.ts [hunk 5]
```diff
@@ -41,6 +42,28 @@ export class NoIconFontIsUsed extends Rule {
...
+        if (identifier.hasOwnProperty("url") && identifier["url"]) {
+          return (
+            $(`link[rel='stylesheet'][href^='${identifier.url}']`).length > 0
+          );
+        }
+      }
+    );
+
+    if (knownExternalStylesheets.length) {
+      return this.fail("It seems like icon fonts are being used on this page.");
```
Comment (sebastianbenz) [inline packages/linter/src/rules/NoIconFontIsUsed.ts:57]: Rephrase to 'Avoid using icon fonts to improve loading speed and accessibility'.

**Review by sebastianbenz — APPROVED** [review]

**Pull request merged**